thumbprint/src/main/java/com/thumbtack/thumbprint/views/edittext/ThumbprintEditTextBase.kt

```diff
@@ -65,7 +67,13 @@ abstract class ThumbprintEditTextBase(
                 ) {
                 }
 
-                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                    hasError = if (errorValidator != null) {
+                        errorValidator?.invoke(s) == true
+                    } else {
+                        hasError
+                    }
+                }
 
                 override fun afterTextChanged(s: Editable?) {
                     updateAndApplyState()
```

Comment:
We probably want to move the validation into the `afterText..` block.

```suggestion
                    if (errorValidator != null) {
                       hasError = errorValidator?.invoke(s) == true
                    }

                    updateAndApplyState()
```